sdks/python/apache_beam/runners/direct/evaluation_context.py

```diff
@@ -69,67 +69,109 @@ class _SideInputsContainer(object):
   to a side input.
   """
 
-  def __init__(self, views):
+  def __init__(self, side_inputs):
     self._lock = threading.Lock()
     self._views = {}
-    self._transform_to_views = collections.defaultdict(list)
+    self._transform_to_side_inputs = collections.defaultdict(list)
+    self._side_input_to_blocked_tasks = collections.defaultdict(list)
 
-    for view in views:
-      self._views[view] = _SideInputView(view)
-      self._transform_to_views[view.pvalue.producer].append(view)
+    for side in side_inputs:
+      self._views[side] = _SideInputView(side)
+      self._transform_to_side_inputs[side.pvalue.producer].append(side)
 
   def __repr__(self):
     views_string = (', '.join(str(elm) for elm in self._views.values())
                     if self._views.values() else '[]')
     return '_SideInputsContainer(_views=%s)' % views_string
 
-  def get_value_or_schedule_after_output(self, side_input, task):
+  def get_value_or_block_until_ready(self, side_input, task, block_until):
+    """Returns the value of a view whose task is unblocked or blocks its task.
+
+    It returns the value of a view whose watermark has been updated and
```

Comment:
The first line of the docstring is a summary. The next too are a more detailed explanation. Then below I write the function's returns. However, I have changed the word 'return' from L90 to 'gets', so that it doesn't sound repetitive.